src/base/bittorrent/session.cpp

```diff
@@ -1191,13 +1191,19 @@ void Session::processBannedIPs(lt::ip_filter &filter)
 void Session::adjustLimits(lt::settings_pack &settingsPack) const
 {
     // Internally increase the queue limits to ensure that the magnet is started
-    const int maxDownloads = maxActiveDownloads();
-    const int maxActive = maxActiveTorrents();
+    // and also check whether the result is valid (i.e. no integer overflows)
+    const auto adjustLimit = [this](const int limit) -> int
+    {
+        if ((limit > -1)
```

Comment:
```c++
if ((limit <= -1)
    return limit;
if ((std::numeric_limits<int>::max() - limit) < m_extraLimit)
    return std::numeric_limits<int>::max();
return (limit + m_extraLimit);
```

Reply:
PR updated.
I slightly reordered the code and along with the new comment I find it easier to understand.